internal/provider/networks_appliance_vlans_resource.go

```diff
@@ -0,0 +1,654 @@
+	object56.SetCidr(data.Cidr.ValueString())
+	object56.SetId(data.VlanId.ValueString())
+	object56.SetApplianceIp(data.ApplianceIp.ValueString())
+	object56.SetGroupPolicyId(data.GroupPolicyId.ValueString())
+	object56.SetMask(int32(data.Mask.ValueInt64()))
+	object56.SetName(data.Name.ValueString())
+	object56.SetSubnet(data.Subnet.ValueString())
+	object56.SetTemplateVlanType(data.TemplateVlanType.ValueString())
+	ipv6 := openApiClient.NewNetworksNetworkIdApplianceSingleLanIpv6()
+	ipv6.SetEnabled(data.Ipv6.Enabled.ValueBool())
+	var prefixAssignments []openApiClient.NetworksNetworkIdApplianceSingleLanIpv6PrefixAssignments
+	for _, prefixAssignment := range data.Ipv6.PrefixAssignments {
+		originInterfaces := []string{}
+		for _, originInterface := range prefixAssignment.Origin.Interfaces.Elements() {
+			originInterfaces = append(originInterfaces, originInterface.String())
+		}
+		prefixAssignments = append(prefixAssignments, openApiClient.NetworksNetworkIdApplianceSingleLanIpv6PrefixAssignments{
+			Autonomous:         prefixAssignment.Autonomous.ValueBoolPointer(),
+			StaticPrefix:       prefixAssignment.StaticPrefix.ValueStringPointer(),
+			StaticApplianceIp6: prefixAssignment.StaticApplianceIp6.ValueStringPointer(),
+			Origin: &openApiClient.NetworksNetworkIdApplianceSingleLanIpv6Origin{
+				Type:       prefixAssignment.Origin.Type.ValueString(),
+				Interfaces: originInterfaces,
+			},
+		})
+	}
+	ipv6.SetPrefixAssignments(prefixAssignments)
+	object56.SetIpv6(*ipv6)
+
+	_, httpResp, err := r.client.ApplianceApi.CreateNetworkApplianceVlan(ctx, data.NetworkId.ValueString()).CreateNetworkApplianceVlan(*object56).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to update resource",
+			fmt.Sprintf("%v\n", err.Error()),
+		)
+	}
+
+	// collect diagnostics
+	if httpResp != nil {
+		tools.CollectHttpDiagnostics(ctx, &resp.Diagnostics, httpResp)
+	}
+
+	// Check for API success response code
+	if httpResp.StatusCode != 204 {
+		resp.Diagnostics.AddError(
+			"Unexpected HTTP Response Status Code",
+			fmt.Sprintf("%v", httpResp.StatusCode),
+		)
+	}
+
+	// Check for errors after diagnostics collected
+	if resp.Diagnostics.HasError() {
+		resp.Diagnostics.AddError("Plan Data", fmt.Sprintf("\n%s", data))
+		return
+	}
+
+	// save inlineResp data into Terraform state.
+	if err = json.NewDecoder(httpResp.Body).Decode(data); err != nil {
+		resp.Diagnostics.AddError(
+			"JSON Decode issue",
+			fmt.Sprintf("%v", httpResp.StatusCode),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+
+	// Write logs using the tflog package
+	tflog.Trace(ctx, "removed resource")
+
+}
+
+func (r *NetworksApplianceVlansResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("network_id"), req.ID)...)
```

Comment:
import should use to id's one is network id and other one is vlanid. check admin resource for import example